packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/Controls.tsx

```diff
@@ -3,42 +3,20 @@ import React from 'react';
 import { useControl } from '../../context/controls';
 import { StorageBrowserElements } from '../../context/elements';
 
-import { CLASS_BASE } from '../constants';
-import { Controls, NavigateItem } from '../Controls';
+import { Controls } from '../Controls';
 import { CommonControl } from '../types';
+import { FileItem } from '../../context/types';
 
-const { Button } = StorageBrowserElements;
+import { useHandleUpload } from './useHandleUpload';
 
-const { Summary, Title: TitleElement } = Controls;
+const { ActionCancel, ActionStart, Summary, Title: TitleElement } = Controls;
 
 export interface LocationActionViewControls<
   T extends StorageBrowserElements = StorageBrowserElements,
 > extends Pick<Controls<T>, CommonControl | 'Summary' | 'Title'> {
   (): React.JSX.Element;
 }
 
-export const Navigate = (): React.JSX.Element => {
-  const [, handleUpdateState] = useControl({ type: 'ACTION_SELECT' });
-
-  return (
-    <NavigateItem.Button
-      className={`${CLASS_BASE}__navigate__container`}
-      onClick={() => {
-        handleUpdateState({ type: 'EXIT' });
-      }}
-    >
-      Exit
-    </NavigateItem.Button>
-  );
-};
-
-export function Primary(props: {
-  children?: React.ReactNode;
-  onClick?: React.MouseEventHandler<HTMLButtonElement>;
-}): React.JSX.Element {
-  return <Button {...props} className={`${CLASS_BASE}__refresh`} />;
-}
```

Comment:
`Primary` should handle `action` begin, cancellation and "done"/exit once the `action` has completed, can we update `ActionStart` to reflect those use cases?

`Navigate` is used for just exiting/closing the `LocationActionView`, think that is what you are addressing with `ActionCancel`? Don't feel strongly abt the `Navigate` name here (might be somewhat misleading vs `Navigate` in the `LocationDetailView`) but think we should avoid naming it `ActionCancel` since the naming implies a use case that is covered by `Primary`. Thoughts?

Reply:
Yeah, I don't think we should re-use the Navigate name/control since it will likely have unique styling for it's use in the breadcrumbs-type control. So it's something that would need it's own variant or way to theme separately. But yes, that is what is meant to be addressed by ActionCancel (maybe ActionExit?). Would exiting the action view also cancel the action if the action is in progress?

For `Primary`, is the idea that it's a single Button that will change it's children/props/use case depending on the state?

Reply:
Also wondering if instead of "Cancel", the exit button would just say "<- Back" (and handle EXIT, or in the case that the action has started it will EXIT and cancel). Then, the `Primary` control wouldn't need to handle exit because it's kind of weird/doesn't make sense having a button that says "Done" and does something. Probably when it's "done" it can convert to a disabled state.

Reply:
After our discussion, renamed ActionCancel -> Exit and ActionStart -> Primary